src/mlir/verona-mlir.cc

```diff
@@ -14,107 +13,111 @@
 #include "mlir/InitAllDialects.h"
 #include "mlir/InitAllPasses.h"
 
+#include "llvm/Support/CommandLine.h"
 #include "llvm/Support/InitLLVM.h"
 
 namespace
 {
+  namespace cl = llvm::cl;
+  // Input file name
+  static cl::opt<std::string> inputFile(
+    cl::Positional,
+    cl::desc("<input file>"),
+    cl::init("-"),
+    cl::value_desc("filename"));
+
   // Source file types, to choose how to parse
-  enum class SourceKind
+  enum class InputKind
   {
     None,
     Verona,
     MLIR
   };
-
-  // Command line options, with output control
-  struct Opt
+  static cl::opt<enum InputKind> inputKind(
+    "x",
+    cl::init(InputKind::None),
+    cl::desc("Input type"),
+    cl::values(clEnumValN(InputKind::Verona, "verona", "Verona file")),
+    cl::values(clEnumValN(InputKind::MLIR, "mlir", "MLIR file")));
+
+  // Output type, what to emit
+  enum class OutputKind
   {
-    std::string grammar;
-    std::string filename;
-    std::string output;
-    bool mlir = false;
-    bool llvm = false;
+    None,
+    MLIR,
+    LLVM
   };
-
-  // Parse cmd-line and set defaults
-  Opt parse(int argc, char** argv)
+  static cl::opt<enum OutputKind> outputKind(
+    "emit",
+    cl::init(OutputKind::None),
+    cl::desc("Output type"),
+    cl::values(clEnumValN(OutputKind::MLIR, "mlir", "MLIR")),
+    cl::values(clEnumValN(OutputKind::LLVM, "llvm", "LLVM IR")));
```

Comment:
Is this giving us a flag like `-emit mlir`, rather than `-emit-mlir`?  If so, that's different from other LLVM front ends in a slightly confusing way.  Does `cl::opt` have a simple way of providing a pair of options like `-emit-llvm` / `-emit-mlir` and using whichever was the last one specified?

Reply:
AFAIK there is no simple way to do that on a single flag (even in other cli libs). I'd have to create two different options like I had before.

Given that we can't yet emit LLVM anyway, I'd be fine in deferring that for later and just assume we want to always emit MLIR.